PubNubMembership/Sources/Membership+PubNub.swift

```diff
@@ -0,0 +1,526 @@
+    )
+
+    (requestConfig.customSession ?? networkSession)
+      .route(
+        router,
+        responseDecoder: FetchMultipleValueResponseDecoder<PubNubMembership.PartialSpace>(),
+        responseQueue: requestConfig.responseQueue
+      ) { result in
+        completion(result.map { response in
+          (
+            memberships: response.payload.data.compactMap {
+              PubNubMembership(user: .init(id: computedUserId), space: $0)
+            },
+            next: PubNub.Page(
+              next: response.payload.next,
+              prev: response.payload.prev,
+              totalCount: response.payload.totalCount
+            )
+          )
+        })
+      }
+  }
+
+  public func fetchMemberships(
+    spaceId: String,
+    includeCustom: Bool = true,
+    includeUserFields: Bool = false,
+    includeUserCustomFields: Bool = false,
+    includeTotalCount: Bool = false,
+    filter: String? = nil,
+    sort: [PubNub.UserMembershipSort] = [],
+    limit: Int? = 100,
+    page: PubNubHashedPage? = nil,
+    requestConfig: PubNub.RequestConfiguration = .init(),
+    completion: @escaping ((Result<(memberships: [PubNubMembership], next: PubNubHashedPage?), Error>) -> Void)
+  ) {
+    let router = ObjectsMembershipsRouter(
+      .fetchMembers(
+        channelMetadataId: spaceId,
+        customFields: .include(
+          custom: includeCustom,
+          user: includeUserFields,
+          userCustom: includeUserCustomFields
+        ),
+        totalCount: includeTotalCount,
+        filter: filter,
+        sort: sort.map { $0.routerParameter },
+        limit: limit,
+        start: page?.start,
+        end: page?.end
+      ),
+      configuration: requestConfig.customConfiguration ?? configuration
+    )
+
+    (requestConfig.customSession ?? networkSession)
+      .route(
+        router,
+        responseDecoder: FetchMultipleValueResponseDecoder<PubNubMembership.PartialUser>(),
+        responseQueue: requestConfig.responseQueue
+      ) { result in
+        completion(result.map { response in
+          (
+            memberships: response.payload.data.compactMap {
+              PubNubMembership(space: .init(id: spaceId), user: $0)
+            },
+            next: PubNub.Page(
+              next: response.payload.next,
+              prev: response.payload.prev,
+              totalCount: response.payload.totalCount
+            )
+          )
+        })
+      }
+  }
+
+  public func addMemberships(
+    users: [PubNubMembership.PartialUser],
+    to spaceId: String,
+    requestConfig: PubNub.RequestConfiguration = .init(),
+    completion: ((Result<Void, Error>) -> Void)?
+  ) {
+    let router = ObjectsMembershipsRouter(
+      .setMembers(
+        channelMetadataId: spaceId,
+        customFields: nil,
+        totalCount: false,
+        changes: .init(
+          set: users.map { .init(metadataId: $0.user.id, status: $0.status, custom: $0.custom?.flatJSON) },
```

Comment:
@CraigLn VSP endpoint has been extended and now also allow include `status`?

Reply:
Yes, `type` was added to User and Spaces, and `status` was added to User, Spaces, and Memberships.  

Since these fields were added on the Objects V2 those fields will now also exist on Object V2 models and APIs.  There is an include parameter flag for each value (`include=status` and `include=type`) to ensure that field is returned from the APIs, and the SDKs are hardcoded to always include those fields.